### DISCUSSION TITLE
Loading .fbx from code-only

### DISCUSSION THREAD

**Discussion by blair-ahlquist** [stride3d/stride#1743] [Q&A]
What is the most direct approach in loading an .fbx file and creating a `Entity` from code-only? I have tried two approaches:

```
string inputFile = "C:\\Users\\bahlq\\Documents\\3D Models\\PeterHeadSimpleHairMesh.fbx";
MeshConverter importer = new MeshConverter(null);
EntityInfo info = importer.ExtractEntity(inputFile, extractTextureDependencies:true);
// Create Entity from EntityInfo?

FbxAssetImporter imp = new ();
AssetItem[] assets = imp.Import(new UFile(inputFile), imp.GetDefaultParameters(true)).ToArray();

foreach(var assetItem in assets)
{
    if (assetItem.Asset is ModelAsset modelAsset)
    {
        // Create Model from ModelAsset?
    }
}

I am interested in using the Assimp importer eventually.

**Comment by blair-ahlquist** [comment]
Bump. I've browsed through the source code looking for a solution...

**Comment by Doprez** [comment]
if you are already ussing the 3d importer from stride to get the model data then you may want to look at ModelFromFileTemplateGenerator.cs in the Stride.Assets.Presentation project. and get the saved file that should be then able to be created with Content.Load<T>("path"). There was a discussion on how to use that [here](https://discord.com/channels/500285081265635328/500285081265635330/1029021377249947738).

another option is to load from obj which someone already got working [here](https://github.com/jeske/LoadObjTest). I havent looked into this much myself as I dont have the need atm, but these are likely starting points for what you need.

**Reply by Doprez** [reply]
Also adding this as I just found [this](https://github.com/ykafia/StrideGltf/blob/master/GltfImport/LoadGLTF.cs) searching in the discord.
